packages/playwright-test/src/workerRunner.ts

```diff
@@ -337,6 +339,25 @@ export class WorkerRunner extends EventEmitter {
       },
     };
 
+    testInfo.attach = async attachment => {
```

Comment:
@dgozman Is there a better way to do this? It smells a little, but pulling out a shared reference to both `outputPath` and the `attachments` array felt equally gross…so this was the cleanest!

Aside—but may become relevant—why are some shared vars wrapped in a function that's immediately invoked (https://github.com/microsoft/playwright/blob/a51ac39275c33161f6d0f57751730afcce48850c/packages/playwright-test/src/workerRunner.ts#L230)? I'm assuming there is good reason related to scoping and worker processes, but not totally sure and piqued my interest!

Reply:
I think you can just inline the method, similar to `_addStep`.

> why are some shared vars wrapped in a function that's immediately invoked

That's just stylistic, to give some piece of logic a name, and avoid introducing many more variables into the scope.

Reply:
Sweet! Forgot you can reference the object that you're actively building out from with itself…what a wild thing! :)